app/code/Magento/Catalog/Model/CategoryList.php

```diff
@@ -103,7 +115,7 @@ public function getList(SearchCriteriaInterface $searchCriteria)
     private function getCollectionProcessor()
     {
         if (!$this->collectionProcessor) {
-            $this->collectionProcessor = \Magento\Framework\App\ObjectManager::getInstance()->get(
+            $this->collectionProcessor = ObjectManager::getInstance()->get(
                 'Magento\Eav\Model\Api\SearchCriteria\CollectionProcessor'
             );
         }
```

Comment:
Could you please move it to constructor?

Reply:
Done. I assumed this method no longer needed, so I removed it.